ports/raspberrypi/common-hal/watchdog/WatchDogTimer.c

```diff
@@ -37,47 +37,50 @@ void common_hal_watchdog_feed(watchdog_watchdogtimer_obj_t *self) {
 }
 
 void common_hal_watchdog_deinit(watchdog_watchdogtimer_obj_t *self) {
-    if (self->mode == WATCHDOGMODE_RESET) {
-        mp_raise_RuntimeError(translate("WatchDogTimer cannot be deinitialized once mode is set to RESET"));
-    } else {
-        self->mode = WATCHDOGMODE_NONE;
+    if (self->mode == WATCHDOGMODE_NONE) {
+        return;
     }
+    hw_clear_bits(&watchdog_hw->ctrl, WATCHDOG_CTRL_ENABLE_BITS);
+    self->mode = WATCHDOGMODE_NONE;
 }
 
-/*
 void watchdog_reset(void) {
-   common_hal_watchdog_deinit(&common_hal_mcu_watchdogtimer_obj);
+    common_hal_watchdog_deinit(&common_hal_mcu_watchdogtimer_obj);
 }
```

Comment:
this is a behavioral change. before, setting a watchdog and then having your program exit (whether normally or by exception) would cause the microcontroller to reset. I believe I have written code that relies on this.

Reply:
(also it's weird that uncrustify hadn't previously normalized the indentation here, I wouldn't have noticed this otherwise)

Reply:
OK, we need to harmonize the behavior across ports, I can agree with that much. The program I'm thinking of is running on a samd51 (itsy bitsy m4) fwiw.

I believe the preferable behavior is that when the watchdog is not **explicitly** deinitialized by user code, it remains active.

This means the watchdog will be effective when the running program terminates unexpectedly, whether by an exception or not. In the case of a reload exception, and maybe a keyboard interrupt, I could be persuaded otherwise. But in the case of, say, a socket exception, you want the watchdog to fire, rather than sitting at the "waiting to reload" prompt. That's an important purpose of the watchdog, and I'm honestly quite surprised I didn't notice this behavior on the espressif port yet.